### PR TITLE
Rewrite diff check to match example in cmp.Diff docs

### LINKED ISSUE #4664: NUD tests incorrectly report `cmp.Diff`
Per the example in https://godoc.org/github.com/google/go-cmp/cmp#Diff the form should be:
```
if diff := cmp.Diff(want, got); diff != "" {
    t.Errorf("MakeGatewayInfo() mismatch (-want +got):\n%s", diff)
}
```
however, NUD tests (and possibly others) are littered with the reverse, e.g.:
https://github.com/google/gvisor/blob/8fa18e8ecb3e3da2015e5c73db9e46494dcfeff7/pkg/tcpip/stack/neighbor_cache_test.go#L252-L254

Fix this.

@puradox cc @ghanan94

### PR DESCRIPTION
**Pull request by mlevesquedion**
Fixes #4664.

### CODE REVIEW

**Review by puradox — APPROVED** [review]
Thank you for doing this! 🙌

**Comment by mlevesquedion** [comment]
Of course! Happy to lend a hand :)

**Comment by avagin** [comment]
@mlevesquedion sorry for the delay. Could you rebase the PR?

**Comment by mlevesquedion** [comment]
No worries. I've rebased the PR.

**Comment by mlevesquedion** [comment]
There was another merge conflict. I've rebased the PR again.

File: pkg/tcpip/stack/neighbor_cache_test.go [hunk 1]
```diff
@@ -1750,6 +1750,34 @@ func TestNeighborCacheRetryResolution(t *testing.T) {
 	}
 }
 
+// TestNeighborCacheStaticResolution checks that static link addresses are
```
Comment (tamird) [inline pkg/tcpip/stack/neighbor_cache_test.go:1753]: looks like bad conflict resolution. this test is failing.

**Comment by avagin** [comment]
```
$ bazel test pkg/tcpip/stack:stack_test
INFO: Analyzed target //pkg/tcpip/stack:stack_test (96 packages loaded, 7852 targets configured).
INFO: Found 1 test target...
FAIL: //pkg/tcpip/stack:stack_test (see bazel-out/k8-fastbuild/testlogs/pkg/tcpip/stack/stack_test/test.log)
Target //pkg/tcpip/stack:stack_test up-to-date:
  bazel-bin/pkg/tcpip/stack/stack_test_/stack_test
INFO: Elapsed time: 35.493s, Critical Path: 16.57s
INFO: 110 processes: 4 internal, 106 linux-sandbox.
INFO: Build completed, 1 test FAILED, 110 total actions
//pkg/tcpip/stack:stack_test                                             FAILED in 11.9s
  bazel-out/k8-fastbuild/testlogs/pkg/tcpip/stack/stack_test/test.log

INFO: Build completed, 1 test FAILED, 110 total actions
$ cat bazel-out/k8-fastbuild/testlogs/pkg/tcpip/stack/stack_test/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //pkg/tcpip/stack:stack_test
-----------------------------------------------------------------------------
--- FAIL: TestNeighborCacheStaticResolution (0.00s)
    neighbor_cache_test.go:1769: unexpected error from neigh.entry(62726f616463617374, '', _, nil, nil): operation would block
FAIL
```

**Comment by mlevesquedion** [comment]
Sorry about that. The failing test should not be there, I have removed it. Also there were some unintended changes to `go.{mod,sum}` that I have just removed.

File: pkg/tcpip/stack/neighbor_cache_test.go (hunk 1 above)
Comment (mlevesquedion) [reply pkg/tcpip/stack/neighbor_cache_test.go]: Sorry about that. Should be fixed now.

**Pull request merged**